test/aws.test.js

```diff
@@ -0,0 +1,100 @@
+/* global describe it before after */
+
+const assert = require('assert')
+const kmsHelper = require('../lib/kms-helper')
+const AWS = require('aws-sdk-mock')
+const sinon = require('sinon')
+const lambdaTester = require('lambda-tester')
+const handler = require('../index.js').handler
+const event = require('../event.json')
+const KinesisWriter = require('../lib/KinesisWriter')
+
+const getRDSString = (file) => {
+  return require('dotenv').config({ path: `${file}` }).parsed['DISCOVERY_STORE_CONNECTION_URI']
+}
+
+describe('AWS', function () {
+  this.timeout(5000)
+  require('dotenv').config({ path: './.env' })
+
+  describe('kmsHelper util', () => {
+    it('should fail when there is no encrypted key', () => {
+      process.env['DISCOVERY_STORE_CONNECTION_URI'] = ''
+      try {
+        kmsHelper.decryptDbCreds()
+      } catch (e) {
+        assert.equal(e.message, 'Missing DISCOVERY_STORE_CONNECTION_URI env variable; aborting.')
+      }
+    })
+
+    it('should call decrypt the db uri', () => {
+      process.env['DISCOVERY_STORE_CONNECTION_URI'] = getRDSString('./test.deploy.env')
+      return kmsHelper.decryptDbCreds()
+        .then(value => {
+          assert.equal(value, 'postgresql://localhost:5432/discovery')
+        })
+    })
+  })
+
+  describe('Connecting to KMS for DB URI string', () => {
+    var spyDecryptFunc
+
+    before(() => {
+      process.env['DISCOVERY_STORE_CONNECTION_URI'] = getRDSString('./deploy.env')
+      spyDecryptFunc = sinon.spy()
+      AWS.mock('KMS', 'decrypt', spyDecryptFunc)
+    })
+
+    it('should call kms.decrypt', () => {
+      kmsHelper.decryptDbCreds()
+      assert.equal(spyDecryptFunc.calledOnce, true)
+    })
+
+    after(() => AWS.restore('KMS', 'decrypt'))
+  })
+
+  describe('Connecting to Kinesis', () => {
+    require('dotenv').config({ path: './deploy.env' })
+    require('dotenv').config({ path: './.env' })
+    var spyPutRecordsFunc
+
+    before(() => {
+      spyPutRecordsFunc = sinon.stub()
+      AWS.mock('Kinesis', 'createStream', spyPutRecordsFunc)
+      AWS.mock('Kinesis', 'describeStream')
+      AWS.mock('Kinesis', 'putRecords', () => Promise.resolve(1))
+    })
+
+    it('should call Kinesis.putRecords from the KinesisWriter util', () => {
+      // Must return the function since it's an asynchronous promise
+      return (new KinesisWriter())
```

Comment:
@nonword the following two tests fail. I think it's because the `kinesis.putRecords` is called behind a promise, but so is `kms.decrypt` and the above test works. I then found this issue https://github.com/dwyl/aws-sdk-mock/issues/67 and they also run into restoring mocked functions before the promises get resolved. I _think_ it could be because the kmsHelper is exported through regular functions but the kinesisWriter is a Javascript class. Do you have any experience unit testing these two different styles?